activerecord/lib/active_record/autosave_association.rb

```diff
@@ -392,18 +392,22 @@ def save_collection_association(reflection)
               records -= records_to_destroy
             end
 
-            records.each do |record|
+            records.each_with_index do |record, index|
               next if record.destroyed?
 
               saved = true
-
               if autosave != false && (@new_record_before_save || record.new_record?)
                 if autosave
                   saved = association.insert_record(record, false)
-                elsif !reflection.nested?
-                  association_saved = association.insert_record(record)
-                  if reflection.validate?
-                    saved = association_saved
+                else
+                  if !reflection.nested?
+                    validate = reflection.options[:validate] != false
+                    if validate
+                      valid = association_valid?(reflection, record, index)
+                      saved = valid ? association.insert_record(record) : false
```

Comment:
Calling `association_valid?` on the line above validates the record:

https://github.com/rails/rails/blob/36390b6bdc75fca70e7a9d74c6270e3f40e77799/activerecord/lib/active_record/autosave_association.rb#L329

Calling `insert_record` also validates the record:

https://github.com/rails/rails/blob/36390b6bdc75fca70e7a9d74c6270e3f40e77799/activerecord/lib/active_record/associations/collection_association.rb#L376-L382

We should pass `false` as the second argument to `insert_record`, to avoid the unnecessary second validation.